docs/structured-streaming-programming-guide.md

```diff
@@ -1689,6 +1689,25 @@ hence the number is not same as the number of original input rows. You'd like to
 There's a known workaround: split your streaming query into multiple queries per stateful operator, and ensure
 end-to-end exactly once per query. Ensuring end-to-end exactly once for the last query is optional.
 
+### State Store and task locality
+
+The stateful operations store states for events in state stores of executors. State stores occupy resources such as memory and disk space to store the states.
+So it is more efficient to keep a state store provider running in the same executor across different streaming batches.
```

Comment:
Probably explicitly mentioning `microbatches`, or if we still concern about continuous processing, then `during the lifetime of the query`?

Reply:
`batches` would also just work - looks like in further sentences you seem to use `batches` simply.